gym/vector/vector_env.py

```diff
@@ -211,6 +213,27 @@ def seed(self, seed=None):
             "Please use `env.reset(seed=seed) instead in VectorEnvs."
         )
 
+    def _add_info(self, infos: dict, info: dict, env_num: int):
+        for k in info.keys():
+            if k not in infos:
+                info_array, array_mask = self._init_info_array(type(info[k]))
+            else:
+                info_array, array_mask = infos[k], infos[f"_{k}"]
+
+            info_array[env_num], array_mask[env_num] = info[k], True
+            infos[k], infos[f"_{k}"] = info_array, array_mask
+        return infos
+
+    def _init_info_array(self, dtype: type) -> np.ndarray:
+        if dtype not in [int, float, bool]:
```

Comment:
Does this work for numpy type, `int64`, etc? 
Could you add this to the docstring to explain the optimised dtypes

Reply:
Something like this `np.float64(1)` will fall into the `object` category. Do you suggest adding all the NumPy numeric dtypes to the list? Maybe in a more clean way instead of a long list of numpy dtypes + native dtypes

Reply:
Yes, I think we should certainly support numpy dtype. For numpy arrays, I think we are going to have to support them through objects not as actual multi-dimensional arrays